app/controllers/blogs_controller.rb

```diff
@@ -45,9 +46,16 @@ def destroy
 
   def set_blog
     @blog = Blog.find(params[:id])
+    raise ActiveRecord::RecordNotFound if @blog.secret? && !@blog.owned_by?(current_user)
```

Comment:
なるほどー、まずブログの集合全体から find して、その上で「見せてもよいもの」に該当しなければ `ActiveRecord::RecordNotFound` を手動で raise する、ということですね。やりたいことができていそうな点はまず good ですね:+1: その上で、要件通りに動くだけではなく、より Rails らしく保守性が高いとされる書き方に寄せたいです。
まず、Rails の基本原則として、コントローラはなるべく薄く保つ方が良く、モデルやその他のメソッドを呼ぶ程度にした方が好ましいというのがあります。なので、本来はコントローラに書きたくないです。

さらにいうと、if 文に書いているような条件って、「何を見せてよいか」というブログ記事に関する業務知識ですよね。これはコントローラの show アクションではなく Blog モデルが持っている方がオブジェクト指向的にも自然だったりします。

なので Blog モデルの方に「見せて良いものの集合」を返す何かを実装しておいて、ここからはそれに対して find するだけ、という形が好ましいです〜！

別の観点から言うと、find って引数に渡した id に対応するレコードが見つからないときに ActiveRecord::RecordNotFound を上げてくれるんですよね。

[https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find)

> Find by id - This can either be a specific id (1), a list of ids (1, 5, 6), or an array of ids ([5, 6, 10]). If one or more records cannot be found for the requested ids, then ActiveRecord::RecordNotFound will be raised.
> 

なので、その意味でもなにかに対して find させて、見つからなければ自然と ActiveRecord::RecordNotFound が発生するように、というのが正攻法だったりします！（つまりここは出題者からの「`find` が raise する ActiveRecord::RecordNotFound を使ってくれぇ〜」というメッセージだったわけですね🙂）